File: C++/OOPX.cpp
#include<iostream>
using namespace std;

class Arithmatic
{
    public :
        int iNo1;
        int iNo2;

        Arithmatic()
        {
            cout<<"Inside Default Constructor\n";
            iNo1 = 0;
            iNo2 = 0;
        }

        Arithmatic(int A, int B)
        {
            cout<<"Inside Parametrized Constructor\n";
            iNo1 = A;
            iNo2 = B;
        }

        int Addition()
        {
            int iAns = 0;
            iAns = iNo1 + iNo2;
            return iAns;
        }

        int Substraction()
        {
            int iAns = 0;
            iAns = iNo1 - iNo2;
            return iAns;
        }
};

int main()
{
    Arithmatic aObj1(11,10);
    Arithmatic aObj2;

    int iRet = 0;

    iRet = aObj1.Addition();
    cout<<"Addition is : "<<iRet<<"\n";

    iRet = aObj1.Substraction();
    cout<<"Substraction is : "<<iRet<<"\n";

    return 0;
}

File: Java/OOPX.java
class Arithematic
{
    public int iNo1;
    public int iNo2;

    public Arithematic()
    {
        this.iNo1 = 0;
        this.iNo2 = 0;
    }

    public Arithematic(int A, int B)
    {
        this.iNo1 = A;
        this.iNo2 = B;
    }

    public int Addition()
    {
        int iAns = 0;
        iAns = this.iNo1 + this.iNo2;
        return iAns;
    }

    public int Substraction()
    {
        int iAns = 0;
        iAns = this.iNo1 - this.iNo2;
        return iAns;
    }
}

class OOPX
{
    public static void main(String Arr[])
    {
        Arithematic aObj1 = new Arithematic(11,10);     //aObj1(reference) : 8 bytes (memory on heap)

        int iRet = aObj1.Addition();
        System.out.println("Addition is : " + iRet);

        iRet = aObj1.Substraction();
        System.out.println("Substraction is : " + iRet);

        aObj1 = null;
    }
}